Add Laravel Scout guide (#2715)
meilisearch/documentation: learn/cookbooks/laravel-scout.mdx

---
title: Laravel Scout guide — Meilisearch documentation
description: Learn how to use Meilisearch with Laravel Scout.
sidebarDepth: 3
---

# Laravel Scout guide

In this guide, you will learn to setup [Laravel Scout](https://laravel.com/docs/10.x/scout) to use Meilisearch in your Laravel 10 application.

## Prerequisites

Before you start, make sure you have the following installed on your machin:
- PHP
- [Composer](https://getcomposer.org/)

You will also need a Laravel application. If you don't have one, you can create a new one by running the following command:

```sh
composer create-project laravel/laravel my-application
```

## Installing Laravel Scout

Laravel comes with out-of-the-box full-text search capabilities via Laravel Scout.

To enable it, navigate to your Laravel application directory and install Scout via the Composer package manager:

```sh
composer require laravel/scout
```

> After installing Scout, you need to publish the Scout configuration file. You can do this by running the following artisan command:

Reviewer: ```suggestion
After installing Scout, you need to publish the Scout configuration file. You can do this by running the following `artisan` command:
```

```sh
php artisan vendor:publish --provider="Laravel\Scout\ScoutServiceProvider"
```

This command should create a new configuration file in your application directory: `config/scout.php`.

## Configuring the Laravel Scout driver

> Let’s configure Laravel Scout to use the Meilisearch driver. First, install the dependencies required to use Scout with Meilisearch via Composer:

Reviewer: ```suggestion
Now you need to configure Laravel Scout to use the Meilisearch driver. First, install the dependencies required to use Scout with Meilisearch via Composer:
```

```sh
composer require meilisearch/meilisearch-php http-interop/http-factory-guzzle
```

Then, update the environment variables in your `.env` file:

```sh
SCOUT_DRIVER=meilisearch
# Use the host below if you're running Meilisearch via Laravel Sail
MEILISEARCH_HOST=http://meilisearch:7700
MEILISEARCH_KEY=masterKey
```

### Local development

> Laravel’s official Docker development environment, Laravel Sail, comes with a Meilisearch service out-of-the-box. Please note that when running Meilisearch via Sail, Meilisearch’s host is `http://meilisearch:7700`.

Author: A "service" is what Docker compose calls each of the "instances" that are run. I understand the fear for unnecessary confusion. 

But on the other hand, for someone familiar with Docker compose, it's  clear. It's the name that they should use to google or search in the Docker docs

<Capsule intent="note">
This comes from Docker [Bridge network driver](https://docs.docker.com/network/drivers/bridge/#differences-between-user-defined-bridges-and-the-default-bridge) behavior.
</Capsule>

### Running in production

For production use cases, we recommend using a managed Meilisearch via [Meilisearch Cloud](https://www.meilisearch.com/cloud?utm_campaign=laravel&utm_source=docs&utm_medium=laravel-scout-guide). On Meilisearch Cloud, you can find your host URL in your project settings.

<Capsule intent="tip">

> Read our [Meilisearch Cloud quick start](/learn/getting_started/cloud_quick_start?utm_campaign=laravel&utm_source=docs&utm_medium=laravel-scout-guide).

Reviewer: ```suggestion
Read the [Meilisearch Cloud quick start](/learn/getting_started/cloud_quick_start?utm_campaign=laravel&utm_source=docs&utm_medium=laravel-scout-guide).
```

</Capsule>

## Making Eloquent models searchable

> With Scout installed and configured, add the `Laravel\Scout\Searchable` trait to your Eloquent models to make them searchable. 🎉
> 
> This trait will use Laravel’s model observers to keep the data in your model in sync with Meilisearch.

Reviewer: Call me heartless. I removed your emoji.

More seriously, I think these two work better as a single paragraph so it reinforces the connection between "this trait" and the trait you're referring to.

```suggestion
With Scout installed and configured, add the `Laravel\Scout\Searchable` trait to your Eloquent models to make them searchable. This trait will use Laravel’s model observers to keep the data in your model in sync with Meilisearch.
```

In Laravel, you can configure your index settings via the `config/scout.php` file:

```php
<?php

use App\Models\Contact;

return [
	  // Other Scout configuration...

    'meilisearch' => [
        'host' => env('MEILISEARCH_HOST', 'https://edge.meilisearch.com'),
        'key' => env('MEILISEARCH_KEY'),
        'index-settings' => [
            Contact::class => [
                'filterableAttributes' => ['organization_id'],
                'sortableAttributes' => ['name', 'company_name']
            ],
        ],
    ],
];
```

The example above updates Meilisearch index settings for the `Contact` model:

- it makes the `organization_id` field filterable
- it makes the `name` and `company_name` fields sortable

After changing your index settings, you will need to synchronize your Scout index settings.

## Synchronizing your index settings

To synchronize your index settings, run the following command:

```sh
php artisan scout:sync-index-settings
```

## Example usage

> We built an example application to demonstrate how to use Meilisearch with Laravel Scout. It showcases an app-wide search in a CRM (Customer Relationship Management) application.

Reviewer: ```suggestion
You built an example application to demonstrate how to use Meilisearch with Laravel Scout. It showcases an app-wide search in a CRM (Customer Relationship Management) application.
```
